src/core/exec-invoke.c

```diff
@@ -2539,9 +2539,11 @@ static int setup_exec_directory(
                 /* Lock down the access mode (we use chmod_and_chown() to make this idempotent. We don't
                  * specify UID/GID here, so that path_chown_recursive() can optimize things depending on the
                  * current UID/GID ownership.) */
-                r = chmod_and_chown(pp ?: p, context->directories[type].mode, UID_INVALID, GID_INVALID);
-                if (r < 0)
-                        goto fail;
+                if (type != EXEC_DIRECTORY_STATE) {
```

Comment:
This should apply to all exec directories, there shouldn't be any reason to avoid it on others. Also, uid mapping might not be available as it's file system dependent and some are only supported on recent kernels, so you'll need to check first, and fallback to chown if it doesn't work. Using the new mount API to create an anonymous mount is probably the best way to go about it